crates/autopilot/src/domain/settlement/mod.rs

```diff
@@ -80,8 +99,22 @@ pub enum Error {
     Solution(#[from] solution::Error),
     #[error("settlement refers to an auction from a different environment")]
     WrongEnvironment,
-    #[error("connection to the persistence layer failed: {0}")]
+    #[error(transparent)]
     PersistenceConnection(#[from] infra::persistence::Error),
     #[error(transparent)]
     Auction(#[from] infra::persistence::error::Auction),
+    #[error(transparent)]
```

Comment:
I may sound like a broken record, but this type of error handling doesn't scale. We are just defining new error types and keep blowing up higher level structs into a composite nightmare. How is any callsite going to handle these errors except from simply printing them?

I think we should think about the higher level type of errors that a settlement can have:
1. Issues with our persistence (those are currently part of PersistenceConnection, Auction, Winner). Those should likely be retried and are not fatal.
2. Issues with the settlement that make the settlement invalid (currently we also have a benign WrongEnvironment error, but this differentiation only matters if we want to circuit break in the autopilot).

These are the two error types we should express and we should provide conversions from all sub error types into such.

Please don't use blanket `#[from]` conversions. Those are a code smell imo and lead to unhandlable error structs that are not any better than using `anyhow` for all types.

Reply:
Handled in the meantime (you probably reviewed non-latest commit)